server/task/src/main/java/org/apache/james/task/Task.java

```diff
@@ -93,7 +93,7 @@ static Result combine(Result result1, Result result2) {
     Result run() throws InterruptedException;
 
 
-    default String type() {
+    default TaskType type() {
```

Comment:
This looks harmful... What if as a DEV I forget to override this? Let's promote safety against convenience?

I propose introducing a `TestTask` that wraps a runnable: this should allow easy-to-instanciate from lambda tasks for tests while forcing DEVs to specify the type of their tasks.

WDYT?

Reply:
I share your concern. But we only added strong typing to  the return type.
This behavior is there since 01/2018.
Could you create an issue in the backlog for this ?

Reply:
Sure!